pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/main/java/org/apache/pinot/plugin/stream/kafka20/KafkaStreamMessage.java

```diff
@@ -35,4 +35,12 @@ public long getNextOffset() {
     }
     return -1;
   }
+
+  public long getOffset() {
+    if (_metadata != null) {
+      long offset = Long.parseLong(_metadata.getRecordMetadata().get(KafkaStreamMessageMetadata.METADATA_OFFSET_KEY));
+      return offset < 0 ? -1 : offset;
```

Comment:
Why do we need to convert a negative offset to `-1`?

Reply:
I referred the `getNextOffset` function for this. But we can just return offset here. Updated